### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by jacobmsft**
Upgrade ubuntu container to 23.04
Upgrade to .net 7.0
Upgrade python to 3.10
Move from pygithub to ghapi

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04 AS codeql_base
+FROM ubuntu:23.04 AS codeql_base
```
Comment (scovetta) [inline Dockerfile:1]: We should probably go 22.04 (LTS)

File: Dockerfile [hunk 2]
```diff
@@ -47,15 +47,14 @@ RUN cd /tmp && \
     apt-get install -y default-jdk apt-transport-https && \
     apt-get update && \
     rm packages-microsoft-prod.deb
-RUN apt-get install -y dotnet-sdk-3.1
+RUN apt-get install -y dotnet-sdk-7.0
```
Comment (scovetta) [inline Dockerfile:50]: Similar, probably stick to .NET 6.

File: Dockerfile (hunk 1 above)
Comment (jacobmsft) [reply Dockerfile]: pushed

File: Dockerfile (hunk 2 above)
Comment (jacobmsft) [reply Dockerfile]: pushed

**Review by scovetta — APPROVED** [review]

**Pull request merged**